elasticsearch7/search-index/src/main/java/org/vertexium/elasticsearch7/Elasticsearch7SearchIndex.java

```diff
@@ -373,13 +364,58 @@ void addElement(
             bulkUpdateService.flushUntilElementIdIsComplete(element.getId());
         }
 
-        UpdateRequestBuilder updateRequestBuilder = prepareUpdate(
-            graph,
+        IndexInfo indexInfo = addPropertiesToIndex(graph, element, element.getProperties());
+        if (additionalVisibilities != null) {
+            ensureAdditionalVisibilitiesDefined(additionalVisibilities);
+        }
+        Map<String, String> source = buildSourceFromElement(graph, element);
+        if (MUTATION_LOGGER.isTraceEnabled()) {
+            MUTATION_LOGGER.trace(
+                "addElement json: %s: %s",
+                element.getId(),
+                Joiner.on(",").withKeyValueSeparator("=").join(source)
+            );
+        }
+
+        List<String> additionalVisibilitiesParam = additionalVisibilities == null
+            ? Collections.emptyList()
+            : toList(additionalVisibilities);
+        List<String> additionalVisibilitiesToDeleteParam = additionalVisibilitiesToDelete == null
+            ? Collections.emptyList()
+            : toList(additionalVisibilitiesToDelete);
+
+        Map<String, Object> fieldsToSet = getPropertiesAsFields(graph, element.getProperties());
+        if (element instanceof Edge) {
+            Edge edge = (Edge) element;
+            fieldsToSet.put(IN_VERTEX_ID_FIELD_NAME, edge.getVertexId(Direction.IN));
+            fieldsToSet.put(OUT_VERTEX_ID_FIELD_NAME, edge.getVertexId(Direction.OUT));
+            fieldsToSet.put(EDGE_LABEL_FIELD_NAME, edge.getLabel());
+        }
+
+        for (Visibility hiddenVisibility : element.getHiddenVisibilities()) {
```

Comment:
On merge, I'll conflict with you here for the PR I made this morning. We need to also loop through all properties here and set the hidden property if any of them are hidden.